src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+            }
+
+            try
+            {
+                string restEndpointPath = _runtimeConfig.RestGlobalSettings.Path;
+                OpenApiComponents components = new()
+                {
+                    Schemas = CreateComponentSchemas()
+                };
+
+                OpenApiDocument doc = new()
+                {
+                    Info = new OpenApiInfo
+                    {
+                        Version = DOCUMENTOR_VERSION,
+                        Title = DOCUMENTOR_UI_TITLE,
+                    },
+                    Servers = new List<OpenApiServer>
+                    {
+                        new OpenApiServer { Url = $"{restEndpointPath}" }
+                    },
+                    Paths = BuildPaths(),
+                    Components = components
+                };
+                _openApiDocument = doc;
+            }
+            catch (Exception ex)
+            {
+                throw new DataApiBuilderException(
+                    message: "OpenAPI description document generation failed.",
+                    statusCode: HttpStatusCode.InternalServerError,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.OpenApiDocumentGenerationFailure,
+                    innerException: ex);
+            }
+        }
+
+        /// <summary>
+        /// Iterates through the runtime configuration's entities and generates the path object
+        /// representing the DAB engine's supported HTTP verbs and relevant route restrictions:
+        /// Paths including primary key:
+        /// - GET (by ID), PUT, PATCH, DELETE
+        /// Paths excluding primary key:
+        /// - GET (all), POST
+        /// </summary>
+        /// <example>
+        /// A path with primary key where the parameter in curly braces {} represents the preceding primary key's value.
+        /// "/EntityName/primaryKeyName/{primaryKeyValue}"
+        /// A path with no primary key nor parameter representing the primary key value:
+        /// "/EntityName"
+        /// </example>
+        /// <returns>All possible paths in the DAB engine's REST API endpoint.</returns>
+        private OpenApiPaths BuildPaths()
+        {
+            OpenApiPaths pathsCollection = new();
+
+            foreach (string entityName in _metadataProvider.EntityToDatabaseObject.Keys.ToList())
+            {
+                // Entities which disable their REST endpoint must not be included in
+                // the OpenAPI description document.
+                if (_runtimeConfig.Entities.TryGetValue(entityName, out Entity? entity) && entity is not null)
+                {
+                    if (entity.GetRestEnabledOrPathSettings() is bool restEnabled)
+                    {
+                        if (!restEnabled)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
+                // Routes including primary key
```

Comment:
nit: looks like this is excluding and the subsequent call is including

Reply:
good eye